maze_dataset/tokenization/maze_tokenizer.py

```diff
@@ -437,45 +449,218 @@ def clear_cache(self):
 
 @serializable_dataclass(frozen=True, kw_only=True)
 class TokenizerElement(SerializableDataclass, abc.ABC):
```

Comment:
maybe have this come with an `_` prefix too, since it's an abstract class? for consistency